modules/AOS_PDF_Templates/formLetter.php

```diff
@@ -135,42 +140,46 @@ static function DVPopupHtml($module)
-				<tr style="height:10px;"><tr><tr><td colspan="2"><button style=" display: block;margin-left: auto;margin-right: auto" onclick="document.getElementById(\'popupDivBack_ara\').style.display=\'none\';document.getElementById(\'popupDiv_ara\').style.display=\'none\';return false;">Cancel</button></td></tr>
-				</table>
-				</div>
-				
-				<div id="popupDivBack_ara" onclick="this.style.display=\'none\';document.getElementById(\'popupDiv_ara\').style.display=\'none\';" style="top:0px;left:0px;position:fixed;height:100%;width:100%;background:#000000;opacity:0.5;display:none;vertical-align:middle;text-align:center;z-index:9998;">
-				</div>
-				<script>
-					function showPopup(){
-						var ppd=document.getElementById(\'popupDivBack_ara\');
-						var ppd2=document.getElementById(\'popupDiv_ara\');
-						if(ppd!=null && ppd2!=null){
-							ppd.style.display=\'block\';
-							ppd2.style.display=\'block\';
-						}else{
-							alert(\'Error!\');
-						}
-					}
-				</script>';
+            echo '
+            <div id="popupDiv_ara" class="modal fade" style="display: none;">
+               <div class="modal-dialog">
+                  <div class="modal-content">
+                     <div class="modal-header">
+                        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>
+                        <h4 class="modal-title">'.$app_strings['LBL_SELECT_TEMPLATE'].'</h4>
+                     </div>
+                     <div class="modal-body">
+                        <div style="padding: 5px 5px; overflow: auto; height: auto;">
+                           <form id="popupForm" action="index.php?entryPoint=formLetter" method="post">
+                              <table width="100%" class="list view table default" cellspacing="0" cellpadding="0" border="0">
+                                 <tbody>';
+                                 foreach ($templates as $templateid => $template) {
+                                    $js = "$('#popupDiv_ara').modal('hide');var form=document.getElementById('popupForm');if(form!=null){form.templateID.value='" . $templateid . "';form.submit();}else{alert('Error!');}";
+                                    echo '<tr height="20" style="background:#f5f5f5; border:1px solid #fff;">
+                                        <td width="17" valign="center"><a href="#" onclick="' . $js . '"><img src="themes/default/images/txt_image_inline.gif" width="16" height="16" /></a></td>
+                                        <td scope="row" align="left"><b><a href="#" onclick="' . $js . '">' . $template . '</a></b></td></tr>';
+                                }
+                              echo '</tbody></table>
+                              <input type="hidden" name="templateID" value="" />
+                            <input type="hidden" name="module" value="' . $module . '" />
+                            <input type="hidden" name="uid" value="' . $_REQUEST['record'] . '" />
```

Comment:
Here too... Hey @urdhvatech Can you check out why Codacy is failing. What I can see its because you are using $_REQUEST['module'] & $_REQUEST['record'].  Processing form data without nonce verification.
https://app.codacy.com/app/salesagility/SuiteCRM/pullRequest?prid=2945728

Reply:
> 
> 
> Here too... Hey, @urdhvatech Can you check out why Codacy is failing. What I can see its because you are using $_REQUEST['module'] & $_REQUEST['record']. Processing form data without nonce verification.
> https://app.codacy.com/app/salesagility/SuiteCRM/pullRequest?prid=2945728

@samus-aran Actually I haven't changed anything except the Layout (modal popup) So the code $_REQUEST['module'] and $_REQUEST['record'] actually pre-existed in the system.

However, I am not sure on what it will require to get it resolved out of nonce verification.

I have tried to use clean_string() function which could be of help. Let me know if that is good. I have made a recent commit for the same